### ISSUE TITLE
FR - Provide an out-of-the-box usable GenericInterface::Invoker::Module for simple Webhooks

### ISSUE THREAD

**Issue by tgurr** [znuny/Znuny#86]
### Environment
- OS: any
- Browser: any
- Znuny version: 6.0.35

### Expected behaviour
Have a usable `GenericInterface::Invoker::Module` for at least simple Webhooks out-of-the-box.

### Actual behaviour
Some time ago I was searching for a way to feed OTRS information to an Node-RED POST endpoint to route information about new tickets to our Matrix chat. Luckily for Matrix<->Node-RED there's [node-red-contrib-matrixbot](https://www.npmjs.com/package/node-red-contrib-matrixbot). But for OTRS I was out of luck, I came up with a hacky solution in the form of a patch [otrs-6.0.10-nodered-connector.patch](https://github.com/znuny/Znuny/files/6598127/otrs-6.0.10-nodered-connector.txt) which basically copies the `Test.pm` and adds it to `Kernel/Config/Files/XML/GenericInterface.xml` to have it available in the GUI which I then could activate in `GenericInterface -> GenericInterface::Invoker::ModuleRegistration -> GenericInterface::Invoker::Module###NodeREDConnector::NodeREDConnector`:

![otrs_01](https://user-images.githubusercontent.com/414984/120795419-baa33d00-c539-11eb-8ad4-55ec90bdc6b2.png)

This allowed me to configure a `Web-Service`:
![otrs_02](https://user-images.githubusercontent.com/414984/120795833-4cab4580-c53a-11eb-8dfc-6afa699b895b.png)

![otrs_04](https://user-images.githubusercontent.com/414984/120796132-bcb9cb80-c53a-11eb-911c-5f96c73dde61.png)

![otrs_05](https://user-images.githubusercontent.com/414984/120796537-3782e680-c53b-11eb-91da-9a32d5b6a09b.png)
![otrs_06](https://user-images.githubusercontent.com/414984/120796549-3b166d80-c53b-11eb-8249-5bf98e291af5.png)
![otrs_07](https://user-images.githubusercontent.com/414984/120796558-3d78c780-c53b-11eb-8267-1f45263dae86.png)

![otrs_03](https://user-images.githubusercontent.com/414984/120796147-c2171600-c53a-11eb-8392-4d69be1b5475.png)

And have it wired up in Node-RED via:
![nodered_otrs](https://user-images.githubusercontent.com/414984/120796806-8c266180-c53b-11eb-815f-2c82b1caeba2.jpg)
and the `extract data` node containing:
```
msg.payload = "Ticket mit der ID " + msg.payload.TicketID + " wurde erstellt." + " URL: https://otrs.domain.local/otrs/index.pl?Action=AgentTicketZoom;TicketID=" + msg.payload.TicketID;
return msg;
```
to construct a payload to pass to the matrix node to post to the chat.

As you can probably see the only payload I get out of this is the `TicketID`, which makes it impossible to filter the incomming information based on queues, title, or any other data to feed to different channels or simply give more information.

Another issues with this is that it generated an error where the admin always gets an email for every ticket created when the invoker triggers:

```
Sender: OTRS Notifications <otrs@otrs.domain.local>
Subject: OTRS Scheduler Daemon GenericInterface: TicketCreate WebService: NodeREDConnector
Message: Error while performing REST 'POST' request to Controller '/otrs' on Host 'https://nodered.domain.local'. Response code '504'.
```

I'm by no means a perl programmer so I was lucky I even got that far and have something working at all. But I'm completely unable to extend the perl invoker adding more data and fixing the error.

Some products like Elasticsearch hide the ability to send information to webhooks behind a payed licence, others like e.g. Gitea even provide support for having [matrix webhooks](https://github.com/go-gitea/gitea/commit/828a27feeb11474b848be183a8fce2d036dec3cc) directly.

![otrs_08](https://user-images.githubusercontent.com/414984/120798274-68641b00-c53d-11eb-9e5d-be32a0658e3b.png)
https://docs.gitea.io/en-us/webhooks/

That would be awesome of course to also have for Znuny, but just a simple generic one would be great for a start as well.

**Comment by rkaldung** [comment]
@tgurr The feature release will have an Invoker for the GenericInterface which suits your use case (and many more).

**Comment by dennykorsukewitz** [comment]
Hi @tgurr 

I'll give you the Issue until then (rel-6_1_1).
Then please check out the new version and see if your case is solved.

Thanks 🚀

**Comment by rkaldung** [comment]
@tgurr Have you seen that Znuny 6.1 provides a [ticket invoker](https://doc.znuny.org/manual/admin/webservices/invoker/index.html#ticket-generic)? This should definitely provide the data you're looking for.

**Comment by tgurr** [comment]
@rkaldung Yes thanks I've seen it and also followed the changes in git master, just waiting for the ITSM packages to be available so I'm able to upgrade to 6.1.x and try things out. Thanks for pointing to the documentation!
